hawkular-alerts-engine-extensions/hawkular-alerts-aggregated-events-alerter/src/main/java/org/hawkular/alerts/extensions/AggregatedEventsExtension.java

```diff
@@ -159,13 +170,16 @@ private void refresh(Set<DefinitionsEvent> defEvents) {
         if (isEmpty(events)) {
             return events;
         }
-        List<Event> retained = new ArrayList<>();
+        TreeSet<Event> retained = new TreeSet<>();
         TreeSet<Event> filtered = new TreeSet<>();
         for (Event event : events) {
             if (event.getTags().get(TAG_NAME) != null && event.getTags().get(TAG_NAME).equals(TAG_VALUE)
```

Comment:
minor: could use that same trick to simplify condition (and add null check for tags):
if ( event.getTags() != null && TAG_VALUE.equals( event.getTags().get(TAG_NAME) && event.getContext().get(CONTEXT_PROCESSED) == null)